### PR TITLE
feat: setup basic architecture

### LINKED ISSUE #1: set basic architecture
following [this repository](https://github.com/gsainfoteam/ziggle-flutter) structure
- [x] flutter_bloc
- [x] freezed
- [x] slang
- [x] auto_route
- [x] get_it
- [x] injectable

### PR DESCRIPTION
**Pull request by 2paperstar**
close #1
- flutter_bloc (rxdart)
- freezed
- slang (flutter_localizations)
- auto_route
- get_it
- injectable

## Release Notes

- **New Features**
  - Added localization support for login actions, including a new entry for "Login with GIST mail."
  - Introduced a Korean localization file for user interface strings.
  - Implemented a splash page that navigates to the login page upon rendering.
  
- **Improvements**
  - Enhanced dependency injection setup for better management of application services.
  - Established a new routing configuration to streamline navigation within the app.

- **Bug Fixes**
  - Updated `.gitignore` to exclude generated files, improving project cleanliness.

- **Chores**
  - Added several new dependencies for improved functionality and localization support.
  - Introduced a new configuration file for localization parameters.
  - Added static color management for centralized usage across the application.

### CODE REVIEW

**Comment by 2paperstar** [comment]
@coderabbitai review

File: lib/app/di/locator.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import 'package:flutter/foundation.dart';
+import 'package:get_it/get_it.dart';
+import 'package:injectable/injectable.dart';
+
+import 'locator.config.dart';
```
[the remark this answers is not in the document]
Comment (2paperstar) [reply lib/app/di/locator.dart:5]: gitignored

File: lib/app/modules/user/presentation/pages/login_page.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+@RoutePage()
+class LoginPage extends StatelessWidget {
+  const LoginPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const Scaffold();
+  }
+}
```
[the remark this answers is not in the document]
Comment (2paperstar) [reply lib/app/modules/user/presentation/pages/login_page.dart:12]: WIP; will be handled another issue/PR

File: lib/app/modules/user/presentation/pages/login_page.dart [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import 'package:flutter/material.dart';
+
+@RoutePage()
+class LoginPage extends StatelessWidget {
+  const LoginPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const Scaffold();
+  }
```
[the remark this answers is not in the document]
Comment (2paperstar) [reply lib/app/modules/user/presentation/pages/login_page.dart:11]: WIP; will be handled another issue/PR

File: lib/app/modules/splash/presentation/pages/splash_page.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+class _SplashPageState extends State<SplashPage> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      context.router.replaceAll([const LoginRoute()]);
+    });
+  }
```
[the remark this answers is not in the document]
Comment (2paperstar) [reply lib/app/modules/splash/presentation/pages/splash_page.dart:21]: WIP; will be handled another issue/PR

**Comment by 2paperstar** [comment]
@coderabbitai review

**Comment by 2paperstar** [comment]
@coderabbitai full review

**Review by GanghyeonSeo — APPROVED** [review]

**Pull request merged**